### PR TITLE
Custom UI scaling

### PR DESCRIPTION
**Pull request by honzapatCZ**
I suppose everything is right, iam just missing the ability to override size, so maybe make it like a property?

### CODE REVIEW

File: Source/Engine/UI/GUI/CanvasRootControl.cs [hunk 1]
```diff
@@ -54,6 +54,38 @@ public override Control FocusedControl
...
+        /// <summary>
+        /// Helper value to determine the custom scale;
+        /// </summary>
+        public Vector2 CustomScaler
+        {
+            get
+            {
+                Vector2 outVal = Vector2.One;
+                Vector2 size = base.Size;
+                Debug.Log(size);
```
Comment (stefnotch) [inline Source/Engine/UI/GUI/CanvasRootControl.cs:74]: There is a stray `Debug.Log(size)` here and another one a few lines below

Comment (honzapatCZ) [reply Source/Engine/UI/GUI/CanvasRootControl.cs:74]: Ye it's not done and doesn't work 100%

**Comment by stefnotch** [comment]
Relevant issue https://github.com/FlaxEngine/FlaxEngine/issues/448

File: Source/Engine/UI/GUI/CanvasRootControl.cs (hunk 1 above)
Comment (stefnotch) [reply Source/Engine/UI/GUI/CanvasRootControl.cs:74]: Ah, fair enough. You could mark this pull request as a draft then.

**Review by Haridev669 — APPROVED** [review]
This is a great improvement

**Comment by mafiesto4** [comment]
Added **Canvas Scaler** control in https://github.com/FlaxEngine/FlaxEngine/commit/d09dfd1f613c007911d999ac398ba82bc228a5df which implements this feature but also adds more functionalities (constant UI scale, DPI-based scaling, etc.). It should be fairly enough to properly scale game UI. I wrote some docs about it too:: https://github.com/FlaxEngine/FlaxDocs/blob/1.5/manual/ui/controls/canvas-scaler.md

**Pull request closed without merging**